### ISSUE TITLE
Creating Filters using the Keyboard is more complex than need be

### ISSUE THREAD

**Issue by jgough** [opensearch-project/OpenSearch-Dashboards#708]
I create a _lot_ of filters... In Kibana 6, filters were easy to create with just the keyboard, saving time and not having to reach for the mouse.

In Kibana 7 the UI was changed and it made it significantly harder to navigate the filter creation with just the keyboard.  Although there have been some improvements there are still some issues.

Here are the steps for creating a filter for _environment:production_ using mostly the keyboard in Kibana 6:
1. Click Add a filter +
2. Type env
3. Tab to select "environment"
4. Press Tab again to select "is"
5. Type p
6. Press down arrow
7. Press Tab
8. Click Save

Here are the same instructions in Kibana 7/OpenSearch Dashboards:
1. Click + Add filter
2. Type env
3. Press down arrow to highlight "environment"
4. Press Enter to select "environment"
5. Press Tab
6. Press Tab
7. Press down arrow
8. Press Enter to select "is"
9. Type p
10. Press down arrow
11. Press Enter to select "production"
12. Click Save

I know this may seem trivial, but simple things like this can be infuriating if you have to do them every day!

There are two issues here:
1. The tab index of the **ᐯ** dropdown icon has a tab index, and have to tab past this.  It should probably not be included in the tab index?
![image](https://user-images.githubusercontent.com/968442/128308678-2f3c6351-4467-4f3d-ae3b-2a965fd10724.png)

2. Cannot use the tab key to select the currently selected item (like any other standard HTML Select box)
![image](https://user-images.githubusercontent.com/968442/128309277-887ddf51-44aa-4619-b8e3-ddf69ec70520.png)

**Comment by tmarkley** [comment]
@kgcreative thoughts?

**Comment by kgcreative** [comment]
@tmarkley - I think it's worth revisiting the tab + keyboard behavior in EUI input / forms. I suspect this might have been an attempt to improve accessibility, but it's had the unintended side effect of adding a lot of extra keystrokes. We should revisit inputs to make sure they are as aligned to browser default as possible.

**Comment by ahopp** [comment]
I agree with @kgcreative - I think it's worth revisiting. We should make sure that anything we change this round doesn't cause additional oscillations in the future. 

Thanks for raising @jgough!

**Comment by jgough** [comment]
Thanks for considering this.  I know it may sound trivial when looking at the steps and the explanation above but I'm sure if you are used to keyboard shortcuts you get very frustrated when these are taken away!

If there is anything else I can do to make this clearer then let me know.
